### PR TITLE
No score calls if score is not requested

### PR DESCRIPTION
**Pull request by PSeitz**
Those score calls can add add up in cost

### CODE REVIEW

File: src/query/weight.rs [hunk 1]
```diff
@@ -7,12 +7,21 @@ use crate::{DocId, Score, TERMINATED};
 /// `DocSet` and push the scored documents to the collector.
 pub(crate) fn for_each_scorer<TScorer: Scorer + ?Sized>(
     scorer: &mut TScorer,
+    requires_scoring: bool,
```
Comment (fulmicoton) [inline src/query/weight.rs]: Can we not add this parameter to this function and instead create a separate function:

```rust
pub(crate) fn for_each_docset<TDocSet: DocSet + ?Sized>(
    docset: &mut TDocset,
    callback: &mut dyn FnMut(DocId)
) -> { ... }
```

Comment (PSeitz) [reply src/query/weight.rs]: I updated the PR, but it still adds a fake score when `collect()`ing

File: src/query/weight.rs [hunk 2]
```diff
@@ -16,6 +16,16 @@ pub(crate) fn for_each_scorer<TScorer: Scorer + ?Sized>(
     }
 }
 
+/// Iterates through all of the documents matched by the DocSet
+/// `DocSet`.
+pub(crate) fn for_each_docset<T: DocSet + ?Sized>(scorer: &mut T, callback: &mut dyn FnMut(DocId)) {
```
Comment (shikhar) [inline src/query/weight.rs]: probably calling the param `docset` makes more sense than `scorer`

File: src/query/boolean_query/boolean_weight.rs [hunk 3]
```diff
@@ -219,6 +219,24 @@ impl<TScoreCombiner: ScoreCombiner + Sync> Weight for BooleanWeight<TScoreCombin
         Ok(())
     }
 
+    fn for_each_no_score(
+        &self,
+        reader: &SegmentReader,
+        callback: &mut dyn FnMut(DocId),
+    ) -> crate::Result<()> {
+        let scorer = self.complex_scorer(reader, 1.0, &self.score_combiner_fn)?;
```
Comment (shikhar) [inline src/query/boolean_query/boolean_weight.rs]: Does `&DoNothingCombiner` make sense here?

Comment (PSeitz) [reply src/query/boolean_query/boolean_weight.rs]: I think so, but it's not called anyway

Comment (fulmicoton) [inline src/query/boolean_query/boolean_weight.rs]: Can we still use the DoNothingCombiner?

```suggestion
        let scorer = self.complex_scorer(reader, 1.0, &DoNothingCombiner)?;
```

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**